src/video/x11/SDL_x11modes.c

```diff
@@ -97,14 +97,14 @@ static SDL_bool ParseDBusReply(SDL_DBusContext *dbus, DBusMessage *reply, int ty
     return SDL_TRUE;
 }
 
-static void UpdateDisplayContentScale(float scale)
+static void UpdateDisplayContentScale(double scale)
```

Comment:
This seems wrong?

Reply:
Making it double avoids a float conversion. It's a small nit pick, but hey :)

This is the only caller
https://github.com/libsdl-org/SDL/blob/17c459e384901cca8579f623890dd736cc70a263/src/video/x11/SDL_x11modes.c#L155

The argument (`new_scale`) is initialized here and is a double:
https://github.com/libsdl-org/SDL/blob/17c459e384901cca8579f623890dd736cc70a263/src/video/x11/SDL_x11modes.c#L148-L151

Reply:
Doing the early cast in https://github.com/libsdl-org/SDL/pull/9959/commits/e2b120c8cfe886b9bd3a8390114c660e8b543d83